docs/conceptual/core/executor.md

```diff
@@ -2,8 +2,147 @@
 title: Executor
 ---
 
-* Runtime Versioning
+The executor is responsible for dispatching and executing calls into the
+Substrate runtime.
 
-* Native VS Wasm Execution
+## Runtime Execution
 
-* Forkless Runtime Upgrades
\ No newline at end of file
+The Substrate runtime is compiled into a native executable and a WebAssembly
+(Wasm) binary.
+
+The native runtime is included as part of the node executable, while the Wasm
+binary is stored on the blockchain under a well known storage key.
+
+These two representations of the runtime may not be the same. For example: when
+the runtime is upgraded. The executor determines which version of the runtime to
+use when dispatching and executing calls.
+
+### Execution Strategy
+
+Before runtime execution begins, the Substrate client proposes which runtime
+execution environment should be used. This is controlled by the execution
+strategy, which can be configured for the different parts of the blockchain
+execution process. The strategies are:
+
+- `Native`: Execute with native build (if available, WebAssembly otherwise).
```

Comment:
Isn't this strategy equal to NativeElseWasm? I don't understand the difference.

Reply:
The difference is that if the native execution fails, it does _not_ try Wasm execution.

You think it would help to write this explicitly?

Reply:
Ok, I get it now. Native only checks availability of the native build, while NativeElseWasm checks availability and correct execution.
Maybe writing it a bit more explicitly would help, they definitely seem the same on a first read.

Reply:
Updated